client/engine/engine.go

```diff
@@ -415,3 +481,27 @@ func (e *Engine) constructObjectiveFromMessage(id protocols.ObjectiveId, ss stat
 	}
 
 }
+
+// getProposalObjectiveId returns the objectiveId for a proposal.
+func getProposalObjectiveId(p consensus_channel.Proposal) protocols.ObjectiveId {
+	switch p.Type() {
+	case consensus_channel.AddProposal:
+		{
+			const prefix = virtualfund.ObjectivePrefix
```

Comment:
I don't have a concrete suggestion but it would be nicer to push this code down into the `consensus_channel` package. I think we want to keep the `engine` as lean as possible.